pkg/storage/chunk/azure/blob_storage_client.go

```diff
@@ -36,6 +37,19 @@ const (
 	azureUSGovernment = "AzureUSGovernment"
 )
 
+var requestDuration = instrument.NewHistogramCollector(prometheus.NewHistogramVec(prometheus.HistogramOpts{
+	Namespace: "loki",
+	Name:      "azure_blob_request_duration_seconds",
+	Help:      "Time spent doing azure blob requests.",
+	// Latency seems to range from a few ms to a few secs and is
+	// important.  So use 6 buckets from 5ms to 5s.
+	Buckets: prometheus.ExponentialBuckets(0.005, 4, 6),
+}, []string{"operation", "status_code"}))
+
+func init() {
+	requestDuration.Register()
```

Comment:
Nit: we're trying to avoid global registration across the board.

I think you can use promauto too, this way you don't need init.

Reply:
Yeah, I largely copied this from our s3 client :(